apps/arweave/src/ar_data_sync.erl

```diff
@@ -1660,6 +1665,38 @@ get_tx_offset(TXIndex, TXID) ->
 			{error, failed_to_read_offset}
 	end.
 
+get_tx_offset_data_in_range(TXOffsetIndex, TXIndex, Start, End) ->
```

Comment:
looking at `ar_kv:get_range3` it looks like it will only include transactions whose data is fully contained within Start and End. Which sounds like the right approach to me!

Reply:
Fixed a bug [here](https://github.com/ArweaveTeam/arweave/pull/556/commits/aaa996b757a01dff6205c38c3c7d323a02386144) where a TXID won't be discovered if the given range is somewhere in the middle.

The function returns a list of {TXID, StartOffset, EndOffset} where StartOffset, EndOffset are transaction borders (not clipped by the given range) for all TXIDs intersecting the given range.

Reply:
Thanks! That explanation helps! I've suggested updating the comment with your wording.